src/brsTypes/components/RoAppInfo.ts

```diff
@@ -0,0 +1,140 @@
+import { BrsBoolean, BrsString, BrsValue, ValueKind } from "../BrsType";
+import { BrsComponent } from "./BrsComponent";
+import { BrsType } from "..";
+import { Callable, StdlibArgument } from "../Callable";
+import { Interpreter } from "../../interpreter";
+import * as PP from "../../preprocessor";
+
+export class RoAppInfo extends BrsComponent implements BrsValue {
+    readonly kind = ValueKind.Object;
+
+    private readonly manifest = PP.getManifestSync(process.cwd());
+
+    constructor() {
+        super("roAppInfo");
+
+        this.registerMethods({
+            ifAppInfo: [
+                this.getID,
+                this.isDev,
+                this.getVersion,
+                this.getTitle,
+                this.getSubtitle,
+                this.getDevID,
+                this.getValue,
+            ],
+        });
+    }
+
+    toString(parent?: BrsType) {
+        return "<Component: roAppInfo>";
+    }
+
+    equalTo(other: BrsType) {
+        return BrsBoolean.False;
+    }
+
+    /**
+     * Originally returns the app's channel ID or 'dev' for sideloaded applications.
+     * @returns {string} - 'dev'
+     */
+    private getID = new Callable("getID", {
+        signature: {
+            args: [],
+            returns: ValueKind.String,
+        },
+        impl: (interpreter: Interpreter) => {
+            return new BrsString("dev");
+        },
+    });
+
+    /**
+     * Returns true if the application is sideloaded, i.e. the channel ID is "dev".
+     * @returns {boolean} - true
+     */
+    private isDev = new Callable("isDev", {
+        signature: {
+            args: [],
+            returns: ValueKind.Boolean,
+        },
+        impl: (interpreter: Interpreter) => {
+            return BrsBoolean.True;
+        },
+    });
+
+    private getVersion = new Callable("getVersion", {
+        signature: {
+            args: [],
+            returns: ValueKind.String,
+        },
+        impl: (interpreter: Interpreter) => {
+            let manifest = this.manifest;
+
+            let subVersions = ["major_version", "minor_version", "build_version"];
+            let version = "";
+
+            for (let i = 0; i < subVersions.length; i++) {
+                let subVersion = manifest.get(subVersions[i]);
+
+                if (subVersion !== undefined) version += subVersion;
+                if (i < subVersions.length - 1) version += ".";
+            }
```

Comment:
I think we could do this in a more idiomatic way:
```suggestion
            let version = ["major_version", "minor_version", "build_version"]
                .map(key => manifest.get(key))
                .filter(key => !!key)
                .concat(".");
```

Reply:
Hi @lkipke 
Thank you for the suggestion. On Roku, the version is returned split by dots (e.g.: `"3.41.2"`) but with the suggested code, we would get` ""4,3,0,."`. Also, if there are no defined versions, on Roku 2 dots are returned `".."`, with that code it would be one: `"."`. So, I've added small changes to align with Roku.